### ISSUE TITLE
Make taxonomy table

### ISSUE THREAD

**Issue by damianooldoni** [inbo/speciesbim#2]
Create a table called `taxonomy` which will be populated by matching the names in `scientificname` to GBIF Backbone (and their upper classification).

This table should contain an `id` (pk) and the fields returned from GBIF API:

1. `id`: int, primary key
1. `usageKey`: integer, e.g. 8141224
1. `acceptedUsageKey`: integer, e.g. 5219243
1. `scientificName`: char, e.g. 'Canis vulpes Linnaeus, 1758'
1. `canonicalName`: char e.g. 'Canis vulpes`
1. `rank`: 'char e.g. `SPECIES`
1. `status`: char e.g. 'SYNONYM'
1. `confidence`: int 100
1. `matchType`: char, e.g. 'EXACT'
1. `kingdom`: char e.g. 'Animalia'
1. `phylum`: char e.g. 'Chordata'
1. `order`: char e.g. 'Carnivora'
1. `family`: char e.g. 'Canidae'
1. `genus`: char e.g. 'Vulpes'
1. `species`: char e.g. 'Vulpes vulpes'
1. `kingdomKey`: int e.g. 1
1. `phylumKey`: int e.g. 44
1. `classKey`: int e.g. 359
1. `orderKey`: int e.g. 732
1. `familyKey`: int e.g. 9701
1. `genusKey`: 5219234
1. `speciesKey`: 5219243
1. `synonym`: logical e.g. True
1. `class`: char e.g 'Mammalia'

I wrote them in the order as returned by `pygbif.name_bakbone()` function.

@peterdesmet : do you think we should provide a different order?

**Comment by niconoe** [comment]
I am wondering if this table is the right place to store `confidence` and `matchtype`. To me those are properties describing the matching operations, not the taxon itself. As such, they would be better placed alongside `lastCheckDate` in the `scientificName` table. For example, multiple scientific names could be matched to the same GBIF taxon, but with a different confidence. 

(we might even want in the future to separate the properties of the match process and the scientific names in two different tables, but I wouldn't jump to that too early, for the sake of simplicity)

**Comment by damianooldoni** [comment]
sure, you are right! I was too fast in copy pasting...

**Comment by niconoe** [comment]
Other question: if all the table in this data coming from GBIF (and is never updated "manually" in the database), wouldn't be better to use a more explicit name such as 'gbiftaxonomy'?

**Comment by damianooldoni** [comment]
well, it can, indeed. It put the accent on the fact that the info comes from GBIF. It will be only a problem if they in the future decide to change their taxonomy backbone provider. But I don't think it is something we should worry about it now.

**Comment by niconoe** [comment]
okay thanks. Will do for now, we can always change back if needed!

**Comment by niconoe** [comment]
Update: running the new `transform_db.py` script (see #3), now creates a (naive and incomplete) version of this table. `CREATE TABLE` statement to improve as we progress.

**Comment by peterdesmet** [comment]
- I agree with @niconoe that match type and confidence does not belong here. Would just drop for now.
- I prefer `taxonomy` as table name: it is more generic
- Not sure we need all the fields: would drop all the `rankKey` fields
- Place `class` after `order`
- Drop `synonym`: it repeats info

**Comment by niconoe** [comment]
I renamed the table to `taxonomy`, as @peterdesmet suggested. I also agree with the remarks on fields, but couldn't implement yet since most fields are not created yet (I wanted to focus on a good process first before adding all the details and columns).

After my comment yesterday, I moved `match type` and `confidence` to the other (scientificname) table. Since I think it's the correct place, that the work is already done and that it's probably info we'll need to keep sooner or later, I suggest keeping it like that for the moment.
